scripts/dataset_processing/tts/compute_feature_stats.py

```diff
@@ -0,0 +1,185 @@
+    )
+    args = parser.parse_args()
+    return args
+
+
+def _compute_stats(values: List[torch.Tensor]) -> Tuple[float, float]:
+    values_tensor = torch.cat(values, dim=0)
+    mean = values_tensor.mean().item()
+    std = values_tensor.std(dim=0).item()
+    return mean, std
+
+
+def main():
+    args = get_args()
+
+    feature_config_path = args.feature_config_path
+    manifest_path = args.manifest_path
+    audio_dir = args.audio_dir
+    feature_dir = args.feature_dir
+    feature_name_str = args.feature_names
+    mask_field = args.mask_field
+    stats_path = args.stats_path
+
+    if not manifest_path.exists():
+        raise ValueError(f"Manifest {manifest_path} does not exist.")
+
+    if not audio_dir.exists():
+        raise ValueError(f"Audio directory {audio_dir} does not exist.")
+
+    if not feature_dir.exists():
+        raise ValueError(
+            f"Feature directory {audio_dir} does not exist. "
+            f"Please check that the path is correct and that you ran compute_features.py"
+        )
+
+    feature_config = OmegaConf.load(feature_config_path)
+    feature_config = instantiate(feature_config)
+    featurizer_dict = feature_config.featurizers
+
+    print(f"Found featurizers for {list(featurizer_dict.keys())}.")
+    featurizers = featurizer_dict.values()
+
+    feature_names = feature_name_str.split(",")
+    # For each feature, we have a dictionary mapping speaker IDs to a list containing all features
+    # for that speaker
+    feature_stats = {name: defaultdict(list) for name in feature_names}
+
+    entries = read_manifest(manifest_path)
+
+    for entry in tqdm(entries):
+        speaker = entry["speaker"]
+
+        entry_dict = {}
+        for featurizer in featurizers:
+            feature_dict = featurizer.load(manifest_entry=entry, audio_dir=audio_dir, feature_dir=feature_dir)
+            entry_dict.update(feature_dict)
+
+        if mask_field:
+            mask = entry_dict[mask_field]
+        else:
+            mask = None
+
+        for feature_name in feature_names:
+            values = entry_dict[feature_name]
+            if mask is not None:
+                values = values[mask]
+
+            feature_stat_dict = feature_stats[feature_name]
+            feature_stat_dict["default"].append(values)
+            feature_stat_dict[speaker].append(values)
+
+    stat_dict = defaultdict(dict)
+    for feature_name in feature_names:
+        mean_key = f"{feature_name}_mean"
+        std_key = f"{feature_name}_std"
+        feature_stat_dict = feature_stats[feature_name]
+        for speaker_id, values in feature_stat_dict.items():
+            speaker_mean, speaker_std = _compute_stats(values)
+            stat_dict[speaker_id][mean_key] = speaker_mean
+            stat_dict[speaker_id][std_key] = speaker_std
+
+    with open(stats_path, 'w', encoding="utf-8") as stats_f:
+        json.dump(stat_dict, stats_f, indent=4)
```

Comment:
Warning will not matter much since the script will almost instantly overwrite it. But for safety I could add an overwrite an overwrite CLI argument (default False).

Reply:
I feel like we should at the very least prompt the user before potentially overwriting something they prepared already. An overwrite arg is fine too. Either way it'll be quick for the user to change if they do want to overwrite (and much less frustrating than erasing previously computed values).

Reply:
Added an overwrite arg.